packages/commerce-sdk-react/src/hooks/ShopperBaskets/action.ts

```diff
@@ -0,0 +1,357 @@
+   */
+    UpdateShipmentForBasket = 'updateShipmentForBasket',
+    /**
+     * Sets a shipping address of a specific shipment of a basket.
+     * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=updateShippingAddressForShipment} for more information about the API endpoint.
+     * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#updateshippingaddressforshipment} for more information on the parameters and returned data type.
+     */
+    UpdateShippingAddressForShipment = 'updateShippingAddressForShipment',
+    /**
+     * Sets a shipping method to a specific shipment of a basket.
+     * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=updateShippingMethodForShipment} for more information about the API endpoint.
+     * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#updateshippingmethodforshipment} for more information on the parameters and returned data type.
+     */
+    UpdateShippingMethodForShipment = 'updateShippingMethodForShipment',
+    /**
+     * This method allows you to apply external taxation data to an existing basket to be able to pass tax rates and optional values for all taxable line items. This endpoint can be called only if external taxation mode was used for basket creation. See POST /baskets for more information.
+     * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=addTaxesForBasket} for more information about the API endpoint.
+     * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#addtaxesforbasket} for more information on the parameters and returned data type.
+     */
+    AddTaxesForBasket = 'addTaxesForBasket'
+}
+
+/**
+ * A hook for performing actions with the Shopper Baskets API.
+ */
+export function useShopperBasketsAction<Action extends ShopperBasketsActions>(
+    action: Action
+): ActionResponse<Argument<Client[Action]>, DataType<Client[Action]>> {
+    type Arg = Argument<Client[Action]>
+    type Data = DataType<Client[Action]>
+    // Directly calling `client[action](arg)` doesn't work, because the methods don't fully
+    // overlap. Adding in this type assertion fixes that, but I don't understand why. I'm fairly
+    // confident, though, that it is safe, because it seems like we're mostly re-defining what we
+    // already have.
+    // In addition to the assertion required to get this to work, I have also simplified the
+    // overloaded SDK method to a single signature that just returns the data type. This makes it
+    // easier to work with when passing to other mapped types.
+    function assertMethod(fn: unknown): asserts fn is (arg: Arg) => Promise<Data> {
```

Comment:
For those who like me are not familiar with this, see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

Reply:
It looks like we are re-declaring this method in each action hook.

is it possible to DRY this `assertMethod` and re-use this util in all action hooks?

Reply:
I was on the fence about that, since it's pretty small and uses local variables.